File: backend/app/schemas/punch.py
import uuid
from datetime import date, datetime
from typing import Literal, Optional
from pydantic import BaseModel

PunchStatusType = Literal['Open', 'In Progress', 'Resolved']
PunchPriorityType = Literal['Low', 'Medium', 'High', 'Critical']


class PunchItemCreate(BaseModel):
    title: str
    description: Optional[str] = None
    priority: PunchPriorityType = 'Medium'
    status: PunchStatusType = 'Open'
    assigned_to_id: Optional[uuid.UUID] = None
    due_date: Optional[date] = None


class PunchItemUpdate(BaseModel):
    title: str
    description: Optional[str] = None
    priority: PunchPriorityType
    status: PunchStatusType
    assigned_to_id: Optional[uuid.UUID] = None
    due_date: Optional[date] = None


class PunchItemOut(BaseModel):
    id: uuid.UUID
    company_id: uuid.UUID
    project_id: uuid.UUID
    title: str
    description: Optional[str]
    status: str
    priority: str
    assigned_to_id: Optional[uuid.UUID]
    assigned_to_name: Optional[str]
    due_date: Optional[date]
    resolved_at: Optional[datetime]
    created_by_id: uuid.UUID
    created_by_name: str
    created_at: datetime


class PunchSummary(BaseModel):
    total: int
    open: int
    in_progress: int
    resolved: int

File: src/types/punch.ts
export type PunchStatus = 'Open' | 'In Progress' | 'Resolved'
export type PunchPriority = 'Low' | 'Medium' | 'High' | 'Critical'

export const PUNCH_STATUSES: PunchStatus[] = ['Open', 'In Progress', 'Resolved']
export const PUNCH_PRIORITIES: PunchPriority[] = ['Low', 'Medium', 'High', 'Critical']

export interface PunchItem {
  id: string
  companyId: string
  projectId: string
  title: string
  description: string | null
  status: PunchStatus
  priority: PunchPriority
  assignedToId: string | null
  assignedToName: string | null
  dueDate: string | null
  resolvedAt: string | null
  createdById: string
  createdByName: string
  createdAt: string
}

export interface CreatePunchItemData {
  title: string
  description?: string
  priority: PunchPriority
  status: PunchStatus
  assignedToId?: string
  dueDate?: string
}

export interface PunchSummary {
  total: number
  open: number
  inProgress: number
  resolved: number
}
